feat(historical-exports): Use new LemonCalendarRange for date range picking

macobo opened the pull request:
## Problem

Depends on https://github.com/PostHog/posthog/pull/11630

@mariusandra released a new nice date range picker. Let's use that over antd in historical exports!

## Changes

- Automatically refresh public job definitions in historical exports v2
- Use LemonModal in Historical exports configuration
- Extend LemonCalendarRange to support displaying inline and autoapplying
- Use LemonCalendarRange in historical exports configuration

## How did you test this code?

- See tests for plugin-server changes

### Screenshots
![image](https://user-images.githubusercontent.com/148820/188119747-1cab3fc2-dcd4-4c70-9bd1-eb7c240cfc7e.png)
![image](https://user-images.githubusercontent.com/148820/188119780-647e2a3d-4858-495d-8997-0e3a87f57408.png)
![image](https://user-images.githubusercontent.com/148820/188119820-b75c8819-0d9c-413d-ae1b-8f065898668d.png)

macobo committed:
Support updating JOB_SPEC for historic exports V2

macobo committed:
Expect daterange instead of two dates for historical exports v2

macobo committed:
Use LemonModal

macobo committed:
Support date range picking in plugin jobs

macobo committed:
update e2e test payload

mariusandra reviewed (changes requested):
We were [discussing earlier](https://github.com/PostHog/posthog/pull/11519#discussion_r959480085) with @Twixes on how to name these calendar picker components.

The end result we want is a `<LemonDatePicker />` that can be used in a form, and is a button with a popup, [similar to the story here](https://github.com/PostHog/posthog/blob/0d7e51bc91783f7a5de81e1d427c3841119192ac/frontend/src/lib/components/LemonCalendar/LemonCalendarRange.stories.tsx#L14-L43). 

With this end goal in mind, I propose a slightly different way to refactor this component to get the `autoApply` feature without writing `autoApply`.

- `LemonCalendar` - this stays as is
- `LemonDateRangePicker.Inline` - remove the header and footer from `LemonCalendarRange`, and basically force-enable `autoApply`
- `LemonDateRangePicker.Dialog` - the old `LemonCalendarRange`, using the `.Inline` calendar, caches the result before applying, etc
- `LemonDateRangePicker` - a full "input field" for selecting a date range: a button with a popup that opens the `.Dialog`

This same pattern could also be followed to turn `LemonCalendarSelect` into `LemonDatePicker`

If we'd add the proposed new props ("don't show apply", "skip header", etc) and open up that path, I'm just afraid of opening the calendar's source in a year to see what else got added 🙈. Composition should make this more bearable.

If you're heading off, I'd be happy to do the changes, though it shouldn't be much copy/pasting from where this landed.

mariusandra commented on frontend/src/lib/components/LemonCalendar/LemonCalendarRange.tsx:
```diff
@@ -69,18 +85,20 @@ export function LemonCalendarRange({ value, onChange, onClose, months }: LemonCa
+                    )}
+                </div>
+            )}
             <div className="p-2">
```
Without a header/footer, this `p-2` added just enough padding to break the flow, and shouldn't be there.

![2022-09-01 16 50 10](https://user-images.githubusercontent.com/53387/188130117-a4330052-eed1-4429-b0de-1b513e30d02a.gif)

macobo committed:
move LemonCalendarRange to separate component dir

macobo committed:
Separate LemonCalendarRangeInline

Note breaking change: The selected dates selector is now not conditional on no of months shown

macobo committed:
Update storybook

macobo commented:
@mariusandra pushed the refactor (w/o popover pullout). Not quite sure if I'm doing this correctly but there it is.

There's a behavioral change around when we display the date range as text at the bottom. Previously when only displaying a single month, it was hidden but now that context is spread across two components and hard to communicate outside of it.

macobo committed:
Job should use inclusive dates from the UI, reflecting the new datepicker

mariusandra commented on frontend/src/lib/components/LemonCalendarRange/LemonCalendarRangeInline.tsx:
```diff
@@ -0,0 +1,126 @@
+            typeof window === undefined ? WIDTH_OF_ONE_CALENDAR_MONTH * CALENDARS_IF_NO_WINDOW : window.innerWidth
+        return Math.min(Math.max(1, Math.floor(width / WIDTH_OF_ONE_CALENDAR_MONTH)), 2)
+    }
+    const [autoMonthCount, setAutoMonthCount] = useState(getMonthCount())
+    useEffect(() => {
+        const listener = (): void => {
+            if (autoMonthCount !== getMonthCount()) {
+                setAutoMonthCount(getMonthCount())
+            }
+        }
+        window.addEventListener('resize', listener)
+        return () => window.removeEventListener('resize', listener)
+    }, [autoMonthCount])
```
Hmm, this auto month code doesn't work at all, not now on app, nor in this PR. I'll happily fix this after the PR.

However, this could go into a hook `const autoMonthCount = useAutoMonthCount()`, which can be shared between the inline and popup components to retain the previous behaviour regarding showing the selected range.

mariusandra commented on frontend/src/scenes/plugins/edit/interface-jobs/PluginJobConfiguration.tsx:
```diff
@@ -3,14 +3,16 @@ import { PlayCircleOutlined, CheckOutlined, CloseOutlined, SettingOutlined } fro
 import { interfaceJobsLogic } from './interfaceJobsLogic'
 import { LemonInput } from '../../../../lib/components/LemonInput/LemonInput'
 import moment from 'moment'
+import { LemonModal } from '../../../../lib/components/LemonModal'
+import { LemonButton } from '../../../../lib/components/LemonButton'
+import { LemonCalendarRangeInline } from '../../../../lib/components/LemonCalendarRange/LemonCalendarRangeInline'
```
```suggestion
import { LemonModal } from 'lib/components/LemonModal'
import { LemonButton } from 'lib/components/LemonButton'
import { LemonCalendarRangeInline } from 'lib/components/LemonCalendarRange/LemonCalendarRangeInline'
```

mariusandra reviewed (approved):
The changes in the plugin server are too deep for me to efficiently review, but I trust they're good. I added a few comments for the frontend, but nothing blocking.

macobo committed:
Update imports
